src/ch02-02-01-Starknet-devnet-rs.md

```diff
@@ -0,0 +1,370 @@
+| Private key | 0xbc7a73821619a17b4163d7ec1ba7c77b
+| Public key | 0x464919a6c3e82fde52dbdaf647851c98a383b26139fbe4963d4495e1295ad4
+
+| Account address | 0x8dc4c8db1770e27d643ba7afebab2788b21cb5f82fa78d4a2a34f5ec062fe6
+| Private key | 0x3ec7437ddbe2ae79b35cd2ada14603e9
+| Public key | 0x2e5440d5ec97ef60b62299852ddc8850ce21b97e58287c0187ed393719a1d8d
+
+| Account address | 0x177116cf8f576774e62dd2525cd74f1ee2bd395805812f21cdb6b34884bc56
+| Private key | 0x7e187acbe3d88dc0c1a0ac983b9a5f8d
+| Public key | 0x419b331604b8a4f7b8a588951419b9d797625fdfd490b81d87a6a34de5f23c9
+
+| Account address | 0x2bef82fb62abd667ee56dbc752033bd86ec7e40cdc56d6263f50bccb5c21a19
+| Private key | 0xc550cc6c1196edb35fe1f8b5ece1ad13
+| Public key | 0x4629cce9836abb945a5fc14f1d8cf01fbc7de1b1e101bea8c05be07b8c061d3
+
+| Account address | 0x4b016dbad4c706afb5b8591e8e2894526581ad7e47b539e73279629d51bbcf1
+| Private key | 0x4c8995c72b8f08e8b309f334fdea7543
+| Public key | 0x2a78370841df60acbe0f6f7e3a614cdd479bcb366a279422b1894fbc7aab424
+
+| Account address | 0x7170678de4ee6b56e3ff1ca8d2d20360d822b5bd485743a0c80dd06d44a3882
+| Private key | 0x2749bb7c72c988b035b1ef312cf3d7ef
+| Public key | 0xb7fe32817253d389251e80c1611b639e5e1cceb5761d1127d93ddd7a651304
+
+| Account address | 0x3a77277569bb8225f7aa3d7dc4e467296c9143f4f778d63ff6189d359f0c409
+| Private key | 0x87a4ed16fad5dac9988d1b2e6cd8872d
+| Public key | 0x29cb629b8d79f1efb928f553a781518ae1389c55fb61443e6087945b35fef4c
+
+| Account address | 0x49f78b29f6eee314a6c297f7a2f943a0c93dc4ec2c8f3c2089c05f351d0660b
+| Private key | 0x94de45f28d52b5808ef42bd8e5261f90
+| Public key | 0x5c7f51e81c5d1770201f6ef31e46fed058b10322478f4f1d57ae4c542a5d60e
+
+Predeployed accounts using class with hash: 0x61dac032f228abef9c6626f995015233097ae253a7f72d68552db02f2971b8f
+Initial balance of each account: 1000000000000000000000 WEI and FRI
+Seed to replicate this account sequence: 534956567
+
+## Crafting a Starknet Smart Contract
+
+**Important:** Before we proceed with this example, please ensure that the versions of both `Scarb` and `starkli` match the specified versions provided below.
+You can install the latest [scarb](#https://docs.swmansion.com/scarb/download.html) version.
+
+```console
+    rustc --version  #1.73.0 (cc66ad468)
+    Scarb --version  #  2.5.0 (c531a6e50)
+    starkli --version  # 0.1.20 (e4d2307)
+```
+
+For a more optimized and faster performance (though with a longer compilation time), run with:
+
+```
+$ cargo run --release
+```
+
+Now begin by initiating a Scarb project:
+
+```bash
+scarb new my_contract
+```
+
+## Configure Environment Variables and the `Scarb.toml` File
+
+Review the `my_contract` project. Its structure appears as:
+
+```bash
+    src/
+      lib.cairo
+    .gitignore
+    Scarb.toml
+```
+
+Amend the `Scarb.toml` file to integrate the `starknet` dependency and introduce the `starknet-contract` target:
+
+```toml
+    [dependencies]
+    starknet = ">=2.3.0"
```

Comment:
Can we work with the latest version of starknet? `2.5.4`

Reply:
sure